blockchain-etl/polygon-etl
# Update reqs for composer 2

Update reqs in preparation for upgrade to Composer 2.

Reqs:
- Dev
  - loosen reqs since these are fairly optional
- Test
  - pin `jsonschema` to match Composer 2 / Airflow 2
  - bump `pytest` and `tox` to latest versions
- Cli
  - pin blockchain reqs (inc. `base58`) to prevent pip conflict surprises
  - leave non-blockchain reqs loose but compatible with Composer 2
  - note that `click` and `requests` are already installed in Composer 2 by default

Cli:
- remove support for python 3.6
- going forward this will help to reduce dependency conflicts
- bump the package version number

GitHub workflows:
- disable Airflow Test, due to dependency conflicts with python 3.8, e.g. `pandas==0.25.1`

Also:
- rename requirements.txt -> requirements_airflow.txt (for consistency with other repos)
- fix a couple of incorrectly named tests
